pkg/upgrade/handle_upgrade.go

```diff
@@ -93,6 +93,8 @@ func (ctl *Controller) handlePlans(ctx context.Context) error {
 				concurrentNodeNames[i] = upgradenode.Hostname(node)
 			}
 			obj.Status.Applying = concurrentNodeNames[:]
+			isComplete := len(concurrentNodeNames) == 0 && obj.Spec.Concurrency > 0
```

Comment:
The concurrency check here isn't quite accurate, it just ends up causing the plan to never be complete rather than always being complete. Not sure if it's worth retaining?